### PR TITLE
Fix rendering of markdown tables

### LINKED ISSUE #11024: Markdown Tables are Misaligned in Preview
### Check for existing issues

- [X] Completed

### Describe the bug / provide steps to reproduce it

The preview of complex tables in markdown tables do not align columns properly, meaning that it is hard to follow what they actually display. This seems to be caused by tables with content of differing lengths while in a table with multiple columns and rows.

Text table:
```
| Bit(s) | Stage | Name                       | Description                                                                                                                    |
| ------ | ----- | ---------------------------|--------------------------------------------------------------------------------------------------------------------------------|
| 0      | s0    | `check_rs1_dep`            | Enables data dependency check for rs1.                                                                                         |
| 1      | s0    | `check_rs2_dep`            | Enables data dependnecy check for rs2.                                                                                         |
| 2-3    | s0    | `pre_alu_a_select`         | Used to multiplex alu input a's input ahead of time.                                                                           |
| 4-6    | s0    | `pre_alu_b_select`         | Used to multiplex alu input b's input ahead of time.                                                                           |
| 7-9    | s0    | `cmp_op_select`            | Selects a comparison operation for branching.                                                                                  |
| 10     | s1    | `mem_in_use`               | Signals that memory is used by this operation, thus stalling the pipeline.                                                     |
| 11-14  | s1    | `alu_op_select`            | Selects an ALU operation.                                                                                                      |
| 15     | s2    | `mem_write_enable`         | Enables the write signal for memory.                                                                                           |
| 16     | s2    | `alu_out_to_mem_addr`      | Toggles between connecting the alu output or the program counter to the memory address.                                        |
| 17     | s2    | `jump_if_branch`           | Executes a jump if the branch condition is true.                                                                               |
| 18-19  | s2    | `pre_writeback_select`     | Used to multiplex the memory input ahead of time.                                                                              |
| 20     | s3    | `reg_write_enable`         | Enables the write signal for the registers.                                                                                    |
| 21     | s3    | `use_pre_wb_over_mem_data` | Toggles between using the multiplexed `pre_writeback` and using the output memory data as an input to the register input data. |
```

Small window screenshot:
Screenshot from 2024-04-25 21-50-07

Large window screenshot:
Screenshot from 2024-04-25 21-50-54

### Environment

Zed: v1.0.0 (Zed Dev ab6b7ba62ceb4af01403ef5c2baf1782deb935f5)
OS: Linux 1.0.0
Memory: 31.3 GiB
Architecture: x86_64

### If applicable, add mockups / screenshots to help explain present your vision of the feature

_No response_

### If applicable, attach your `~/Library/Logs/Zed/Zed.log` file to this issue.

_No response_

### PR DESCRIPTION
**Pull request by andarut**
- Closes: https://github.com/zed-industries/zed/issues/11024

## Release Notes:

- Improved Markdown Preview rendering of tables

## Before:

image

Screenshot 2024-09-25 at 05 47 19

## Now:

image

Screenshot 2024-09-25 at 05 47 48

## Note:

I'm not a Rust programmer and this is my first PR in Zed (because i just want to fix this, so i can view my notes in Markdown in Zed, not slow Visual Studio Code) -  so there may be errors. I'm open for critic a

### CODE REVIEW

**Comment by andarut** [comment]
fix formatting, how `cargo fmt --all -- --check` passes

File: crates/client/Cargo.toml [hunk 1]
```diff
@@ -55,6 +55,10 @@ url.workspace = true
 util.workspace = true
 worktree.workspace = true
 
+# fix for https://github.com/zed-industries/zed/actions/runs/11025748924/job/30632735188?pr=18315
+[package.metadata.cargo-machete]
+ignored = ["fs"]
+
```
Comment (MrSubidubi) [inline crates/client/Cargo.toml:61]:
```suggestion
```

This is wrong, you should revert the commit and instead merge the `main`-branch into your branch, as #18324 was merged earlier and resolves this issue.

Comment (andarut) [reply crates/client/Cargo.toml]: ok, done

**Comment by andarut** [comment]
i think it's because table starts resizing from left columns, so if you put preview to the left and right side you will see different resizing.

i'm thinking about using absolute size in rems, and in last column use rest of the screen

**Comment by andarut** [comment]
@Angelk90 how i can get current window width and height, so i can resize table with my own resizing?

**Comment by mikayla-maki** [comment]
This is a good improvement, thank you!
